src/char_mapping.rs

```diff
@@ -187,7 +285,44 @@ impl Encoding4 {
 
 impl Encoding6 {
     /// Return the mapping.
-    pub fn mapping(&self) -> HashMap<u16, u16> {
-        unimplemented!();
+    pub fn mapping(&self) -> HashMap<u16, GlyphID> {
+        let mut map = HashMap::new();
+        for (i, glyph_id) in self.glyph_ids.iter().enumerate() {
+            map.insert(self.first_code + i as u16, *glyph_id);
+        }
+        map
+    }
+}
+
+impl Encoding12 {
+    /// Return the mapping.
+    pub fn mapping(&self) -> HashMap<u32, GlyphID> {
+        let mut map = HashMap::new();
+        for group in &self.groups {
+            for i in 0..(group.end_char_code - group.start_char_code + 1) {
+                map.insert(
+                    group.start_char_code + i,
+                    group.start_glyph_id as u16 + i as u16,
+                );
+            }
+        }
+        map
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::io::Cursor;
+
+    use Tape;
+    use super::VariationSelector;
+
+    #[test]
+    fn variation_selector_record() {
+        let mut buf = Cursor::new(vec![0x02u8, 0x01, 0xFF, 0x00, 0x02, 0x01, 0xFF, 0xAA, 0x02, 0x01, 0xFF]);
+        let vsr = buf.take::<VariationSelector>().unwrap();
+        assert!(vsr.character == 0x000201FF);
+        assert!(vsr.default_uvs_offset == 0x000201FF);
+        assert!(vsr.non_default_uvs_offset == 0xAA0201FF);
```

Comment:
By the way, why don’t you use `assert_eq!`?

Reply:
To be honest I didn't think about it here but `assert_eq!` is [getting deprecated](http://rust-lang.github.io/rfcs/2011-generic-assert.html#how-we-teach-this) and [clippy stopped linting this](https://github.com/rust-lang-nursery/rust-clippy/issues/2090). I can still change it if you would like.

Reply:
I wasn’t aware of this deprecation. Then keep it as it is.